pkg/db-repo/repository.go

```diff
@@ -27,6 +27,28 @@ const (
 var operations = []string{Create, Read, Update, Delete, Query}
 var RecordNotFound = gorm.ErrRecordNotFound
```

Comment:
we should try to remove the usage of the `RecordNotFound` error and replace it with the new custom error..